### PR TITLE
Add merlin6 at PSI

### PR DESCRIPTION
**Pull request by superstar54**
In this PR, I added codes and computers for merlin6 at PSI. For more info on Merlin6 at PSI, please visit: https://lsm-hpce.gitpages.psi.ch/merlin6/connect-from-linux.html

### Computer
- cpu
- gpu

### Code
- pw-7.0-cpu
- pw-7.0-gpu
- dos-7.0-cpu
- projwfc-7.0-cpu
- pp-7.0-cpu

### CODE REVIEW

File: merlin.psi.ch/gpu/computer-setup.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+label: merlin6
+hostname: merlin-l-01.psi.ch
+description: Merlin6 HPC at PSI
+transport: ssh
+scheduler: slurm
+work_dir: /home/{username}/aiida_run/
+shebang: '#!/bin/bash'
+mpirun_command: srun -n {tot_num_mpiprocs}
+mpiprocs_per_machine: 1
```
Comment (unkcpz) [inline merlin.psi.ch/gpu/computer-setup.yaml:10]: This is the default number of mpiprocs, I don't think they only have 1 process.

File: merlin.psi.ch/cpu/computer-setup.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+label: merlin6
+hostname: merlin-l-01.psi.ch
+description: Merlin6 HPC at PSI
+transport: ssh
+scheduler: slurm
+work_dir: /home/{username}/aiida_run/
+shebang: '#!/bin/bash'
+mpirun_command: srun -n {tot_num_mpiprocs}
+mpiprocs_per_machine: 1
```
Comment (unkcpz) [inline merlin.psi.ch/cpu/computer-setup.yaml:10]: Same here, the `1` seems not correct.

File: merlin.psi.ch/cpu/computer-setup.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+label: merlin6
+hostname: merlin-l-01.psi.ch
+description: Merlin6 HPC at PSI
+transport: ssh
+scheduler: slurm
+work_dir: /home/{username}/aiida_run/
```
Comment (unkcpz) [inline merlin.psi.ch/cpu/computer-setup.yaml:7]: Can you confirm that they don't provide `/scratch` folder?

File: merlin.psi.ch/codes/pw-7.0-gpu.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+label: pw-7.0
+description: PW compiled for GPU
```
Comment (unkcpz) [inline merlin.psi.ch/codes/pw-7.0-gpu.yaml:3]: why only GPU setup for `pw.x`?

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks @superstar54, some minor requests and questions, otherwise look all good!

My plan is after we merge this we give it a test on the PSI AiiDAlab deployment (http://dev-aiidalab.psi.ch). It is not easy to find all the issues on configuration files without actual test.

File: merlin.psi.ch/cpu/computer-setup.yaml (hunk 3 above)
Comment (superstar54) [reply merlin.psi.ch/cpu/computer-setup.yaml:7]: Thanks! There are two scratch folders:
- /scratch, for jobs running in one node.
- /shared-scratch, for jobs running on multiple nodes.
I change it to `/shared-scratch`.

**Review by unkcpz — APPROVED** [review]

**Pull request merged**